### PR TITLE
Ensure API client headers are set

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1787" title="DX-1787" target="_blank">BugDX-1787</a>  State Tool should send its User Agent string to solver request
  </td></table>
  <br />

### CODE REVIEW

File: pkg/platform/errors.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/rollbar"
+)
+
+type ErrCountryBlocked struct{ *locale.LocalizedError }
+
+func NewCountryBlockedError() *ErrCountryBlocked {
+	rollbar.DoNotReportMessages.Add(locale.T("err_country_blocked"))
+	return &ErrCountryBlocked{LocalizedError: locale.NewInputError("err_country_blocked")}
+}
```
Comment (MDrakos) [inline pkg/platform/errors.go:13]: This was moved from it's original location so I could import `retryhttp` in the `api` package without an import cycle.

File: internal/gqlclient/gqlclient.go [hunk 2]
```diff
@@ -53,7 +55,11 @@ func NewWithOpts(url string, timeout time.Duration, opts ...graphql.ClientOption
 }
 
 func New(url string, timeout time.Duration) *Client {
-	return NewWithOpts(url, timeout, graphql.WithHTTPClient(retryhttp.DefaultClient.StandardClient()))
+	httpClient := api.NewHTTPClient()
+	if condition.InUnitTest() {
+		httpClient = http.DefaultClient
+	}
+	return NewWithOpts(url, timeout, graphql.WithHTTPClient(httpClient))
```
Comment (MDrakos) [inline internal/gqlclient/gqlclient.go:62]: For some reason, the `retryhttp` client is having trouble connecting with the `platform.activestate.testing.tld` URL in unit tests. I spent some time trying to determine why and then just decided to go with this fix.

Comment (Naatan) [reply internal/gqlclient/gqlclient.go:62]: Wouldn't this be better addressed inside `api.NewHTTPClient()`?

File: pkg/platform/errors.go (hunk 1 above)
Comment (Naatan) [reply pkg/platform/errors.go:13]: Are you sure this is still relevant? I don't see this being called from `retryhttp`.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: pkg/platform/errors.go (hunk 1 above)
Comment (MDrakos) [reply pkg/platform/errors.go:13]: It's being used in a different file but same package: https://github.com/ActiveState/cli/blob/DX-1787/internal/retryhttp/roundtripper.go#L33

File: internal/gqlclient/gqlclient.go (hunk 2 above)
Comment (MDrakos) [reply internal/gqlclient/gqlclient.go:62]: Yes, that's a much better place for it.

**Comment by MDrakos** [comment]
This will also fix https://activestatef.atlassian.net/browse/DX-1825

**Review by Naatan — APPROVED** [review]

**Pull request merged**